fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ReduceCastRule.java

```diff
@@ -158,7 +158,7 @@ public boolean checkCastTypeReduceAble(Type parent, Type child, Type grandChild)
 
         // cascaded cast cannot be reduced if middle type's size is smaller than two sides
         // e.g. cast(cast(smallint as tinyint) as int)
-        if (parentSlotSize > childSlotSize && childSlotSize < grandChildSlotSize) {
+        if (parentSlotSize >= childSlotSize && childSlotSize <= grandChildSlotSize) {
```

Comment:
there is an edge case, `parentSlotSize == childSlotSize == grandChildSlotSize`, will this is good to do the cascade cast reduction or not?